src/cmd/singularity/cli/sign.go

```diff
@@ -17,26 +17,43 @@ import (
 
 func init() {
 	SignCmd.Flags().SetInterspersed(false)
+	SignCmd.Flags().StringVarP(&keyServerURL, "url", "u", "", "specify the key server URL (default: https://keys.sylabs.io)")
 	SingularityCmd.AddCommand(SignCmd)
 }
 
 // SignCmd singularity sign
 var SignCmd = &cobra.Command{
 	DisableFlagsInUseLine: true,
-	Args:   cobra.ExactArgs(1),
+	Args:   cobra.RangeArgs(1, 2),
 	PreRun: sylabsToken,
 
 	Run: func(cmd *cobra.Command, args []string) {
 		// args[0] contains image path
 		fmt.Printf("Signing image: %s\n", args[0])
-		if err := signing.Sign(args[0], authToken); err != nil {
+		if err := doSignCmd(args[0], keyServerURL); err != nil {
 			sylog.Errorf("signing container failed: %s", err)
 			os.Exit(2)
 		}
+		fmt.Printf("Signature created and applied to %v\n", args[0])
 	},
 
 	Use:     docs.SignUse,
 	Short:   docs.SignShort,
 	Long:    docs.SignLong,
 	Example: docs.SignExample,
 }
+
+func doSignCmd(cpath, url string) error {
+	if url == "" {
+		// lookup key management server URL from singularity.conf
```

Comment:
Overriding with the `singularity.conf` value here might be confusing to the user. Let's say I run `singularity sign -h`, and I see:

```
  -u, --url string   specify the key server URL (default:
                     https://keys.sylabs.io)
```

So now I think, OK, that default is not what I want. I want it to go to `https://otherkeys.sylabs.io`, so I do `singularity sign -u https://otherkeys.sylabs.io`. Now, unbeknownst to me, `singularity.conf` has `https://yetanotherkeys.sylabs.io` in it. I run that, and it sends my key to `https://yetanotherkeys.sylabs.io`.

I think the behaviour I would expect in the above scenario, is that the value in `singularity.conf` would affect the _default_, but then if I explicitly set `-u`, it overrides that default. Does that make sense?

Reply:
Yeah, that logic is not done. Only a comment as a reminder that maybe we wanna setup a list of key servers in singularity.conf. I will remove that entire block for now since default + the usage of -u is functional until we decide on a format to store a list of keys in singularity.conf or a file similar.